tests/python/pants_test/projects/test_testprojects_integration.py

```diff
@@ -34,6 +34,11 @@ def targets(self):
       'testprojects/src/java/org/pantsbuild/testproject/thriftdeptest',
       # TODO(Eric Ayers): I don't understand why this fails
       'testprojects/src/java/org/pantsbuild/testproject/jvmprepcommand:compile-prep-command',
+      # TODO(#6455): this produces a resolution error in (currently) the test_shard_6() method with
```

Comment:
I'm not totally sure, but I'm pretty damn sure, the issue here is `allow_prereleases`. The native dists are always prereleases as things stand with their fingerprint version numbering scheme. Pants via pex (like pip), does not resolve prereleases by default - this has to be turned on. I think you'll need come up with a semi-substantial change to Pants' pex building code / PythonRequirement to allow for isolated resolves of only certain requirements with prereleases even if the overall resolve mode is the default of `allow_prereleases=False`. I dug a bit this am because hello_again resolution failures were blocking my `git commit` via the lint hook failing. I think it makes sense to uninstall this task pipeline by default until its working reliably. If twitter needs it installed, that could be done with a site-specific plugin register.

Reply:
OK - this was tied up in a fairly subtle dance:
1. `allow_prereleases`
2. leaking of `pantsbuild.pants` prereleases from tests into pants requirement resolution cache via:
 https://github.com/pantsbuild/pants/blob/46d5cb9c7a73758678b073800ca41257f6809ac7/tests/python/pants_test/backend/python/pants_requirement_integration_test_base.py#L43
3. `egg_info --tag-build=+[rest]` interacting badly when applied to a prerelease version.

I'll file here shortly with all the details.

Reply:
I suspected something was funky when the version string had the `+` turned into an `_` in the error message. I thought I remembered hearing that `allow_prereleases` was slightly different than the result of using the `--tag-build` option from from someone, but I don't recall exactly -- it would make sense if "prereleases" is a superset of tagged builds.

Reply:
Also, uninstalling this task pipeline (the native backend as well as the `python_dist()` support) by default sounds great, and is the focus of #5970, which I was going to get to after this and another open PR got merged.

Reply:
OK. #5970 is about contrib - I meant more - no matter where this is housed, have it turned off by default, even in pantsbuild/pants - until it is stable. It's caused more than its fair share of thrash due to instability.

Reply:
Sorry, I also didn't make it clear that my intention with #5970 was also absolutely to turn these off by default (I already wanted to do this just because downloading a huge binary compiler distribution alone feels like something I would want to have to opt in to).